Marker docs RC2

aeshky opened the pull request:
**Proposed changes**:
- Marker docs ready for a second round of reviews. 
- Please comment on the old PR https://github.com/RasaHQ/rasa/pull/10095. I will address the changes in this PR. And you can approve once you're satisfied. 

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

koaning reviewed (approved):
As requested, I've added comments on https://github.com/RasaHQ/rasa/pull/10095. I very much like the addition of the paragraphs explaining the merit of the number of user turns. There's some minor suggestions that I was able to spot, most of which a mere details. 

Hitting approve for now, but would request @aeshky to have a quick look at some of the comments.

aeshky commented:
Thanks @koaning. I have addressed your comments [here](https://github.com/RasaHQ/rasa/pull/10228/commits/dd09de29a8f1038cc6d7badaeac93dbcec1498d3) and [here](https://github.com/RasaHQ/rasa/pull/10228/commits/ff3f23c78f6ab93b4ad0a80af9bf6141656e62a9).
